jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -522,21 +522,16 @@ public void configureTempDirectory(File dir, WebAppContext context)
         if (dir == null)
             throw new IllegalArgumentException("Null temp dir");
 
-        //if dir exists and we don't want it persisted, delete it
-        if (dir.exists() && !context.isPersistTempDirectory())
+        // if it doesn't exist make it
+        if (!dir.exists())
         {
-            if (!IO.delete(dir))
-                throw new IllegalStateException("Failed to delete temp dir " + dir);
+            if (!dir.mkdirs())
+            {
+                throw new IllegalStateException("Unable to create temp dir " + dir);
+            }
         }
 
-        //if it doesn't exist make it
-        if (!dir.exists())
-            dir.mkdirs();
-
-        if (!context.isPersistTempDirectory())
-            dir.deleteOnExit();
-
-        //is it useable
+        // is it useable
```

Comment:
You need to keep the if (!context.isPersistTempDirectory) dir.deleteOnExit.

Reply:
That's covered in makeTempDirectory() already.

Reply:
Scratch that, i'm wrong about this specific one.

Reply:
No, it's also called directly by resolveTempDirectory in the case where someone has explicitly supplied the temp directory to use or the javax.servlet.context.tempDir has been set - whether these are persisted or not is optional and configured by configureTempDirectory.

Reply:
deleteOnExit has been restored.